builder/builder.go

```diff
@@ -1838,6 +1838,12 @@ func (b *Builder) AddRPMList(rpms []string) error {
 		if _, err := os.Stat(repoPath); err == nil {
 			continue
 		}
+		// Remove source RPMs because they should not be added to mixes
+		if strings.Contains(rpm, ".src.rpm") {
+			if err := os.RemoveAll(filepath.Join(b.Config.Mixer.LocalRPMDir, rpm)); err != nil {
```

Comment:
Optional: This should probably fail if that's a directory instead of a file, so `os.Remove` would probably be better here.